### PR TITLE
Add items from url to transfer

### PR DESCRIPTION
**Pull request by davidenko87**
## Proposed changes

This adds the possibility to add files from a web url to your transfer

## Types of changes

What types of changes does your code introduce to wetransfer_ruby_sdk?
_Put an `x` in the boxes that apply_

- [x] Docs (missing or updated docs)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Improvement / maintenance (removing technical debt, performance, tooling, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/WeTransfer/wetransfer_ruby_sdk/blob/master/.github/CONTRIBUTING.md) doc
- [x] I have added necessary documentation (if appropriate)
- [x] Code changes are covered by unit tests.
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: lib/we_transfer_client/transfer_builder.rb [hunk 1]
```diff
@@ -16,13 +16,30 @@ def add_file_at(path:)
...
   def add_web_content(path:)
     url = open(path, allow_redirections: :safe).base_uri.to_s
     url_title = url.split('/').last
     @items << FutureWebItem.new(url: url, title: url_title)
     true
   end
 
+  def save_to_file(path:)
+    # allow_redirection is needed to support http -> https redirection
+    url = open(path, allow_redirections: :safe).base_uri.to_s
```
Comment (bermannoah) [inline lib/we_transfer_client/transfer_builder.rb]: Again, I think doing this without anything like format_parser's [read limiter](https://github.com/WeTransfer/format_parser/blob/master/lib/read_limiter.rb) and constrained [remote io](https://github.com/WeTransfer/format_parser/blob/master/lib/remote_io.rb) modules opens us (and our users) up to a whole bunch of issues. We should _at least_ allow for / validate / check for / require a content-length header for a remote file.

File: spec/integration_spec.rb [hunk 2]
```diff
@@ -157,4 +157,78 @@ def broken.read(*)
...
+      add_result = builder.add_file_from_url(path: 'https://cdn.pixabay.com/photo/2017/01/03/02/07/vine-1948358_1280.png')
+      expect(add_result).to eq(true)
+    end
+    expect(transfer).to be_kind_of(RemoteTransfer)
+    expect(transfer.name).to eq('PNG images')
+    expect(transfer.description).to eq('Images from url')
+    expect(transfer.items).to be_kind_of(Array)
+    expect(transfer.items.length).to eq(2)
+  end
+
```
Comment (bermannoah) [inline spec/integration_spec.rb]: Can we add a sad-path here? What will break this method?

File: spec/we_transfer_client/transfer_builder_spec.rb [hunk 3]
```diff
@@ -37,4 +37,14 @@
     expect(item.title).to eq('wetransfer.com')
     expect(item.local_identifier).to be_kind_of(String)
   end
+
+  it 'adds a file from url' do
+    transfer_builder = described_class.new
+    transfer_builder.add_file_from_url(path: 'https://images.pexels.com/photos/1115804/pexels-photo-1115804.jpeg')
```
Comment (bermannoah) [inline spec/we_transfer_client/transfer_builder_spec.rb]: What happens to this test if this URL changes/gets deleted?

**Comment by davidenko87** [comment]
I'll close this PR, downloading files and provide the security that i wanted is not that easy to implement.

**Pull request closed without merging**